### DISCUSSION TITLE
Consider setting `persistent_workers=True` in 'train_dataloader' to speed up the dataloader worker initialization.

### DISCUSSION THREAD

**Discussion by ghost** [open-edge-platform/anomalib#2035] [Q&A]
Anomalib library is installed.

Copy-pasted the code listed in "Training" section of Anomalib webpage. Programming is running however there is no output.
Suggestions on screen suggest setting `persistent_workers=True` in 'train_dataloader' to speed up the dataloader worker initialization. Screen shot attached.

Checking the "Task Manager" display the CPU and Memory consumption by the Python Kernel to be low. Initially at the start of the program the consumption are high however the consumption values falls low.

Appreciate help to locate the python/configuration file for setting `persistent_workers=True`.
![Untitled](https://github.com/openvinotoolkit/anomalib/assets/18460699/b59bc4bd-c48d-4ff8-b233-6c12aa64df54)

**Comment by samet-akcay** [comment]
This is where the dataloaders are initialized. If you want to add `persistent_workers=True`, you could add them here
https://github.com/openvinotoolkit/anomalib/blob/6c9fd4e28f9df3d2aea7fede77ca0b0c77601e08/src/anomalib/data/base/datamodule.py#L215-L242

**Reply by ghost** [reply]
Thanks Samet.

Added a line of code to the train_dataloader function.
            persistent_workers=True,

Suggestion message is not displayed anymore. Code is running. However we are not able to figure out progress in model training. A progress indicator or verbose status messages could help.

Checking the "Task Manager" display the CPU and Memory consumption by the Python Kernel to be low. Initially at the start of the program the consumption are high however the consumption values falls low.
![Untitled](https://github.com/openvinotoolkit/anomalib/assets/18460699/c12e139e-df0e-4b21-b5f8-cd468ae0e137)
.

**Reply by samet-akcay** [reply]
> However we are not able to figure out progress in model training. A progress indicator or verbose status messages could help.

What do you mean by this? What sort of indicator do you want to see?

**Reply by ghost** [reply]
Could it be possible to indicate task completion in percentage.
